Added action 'woocommerce_variation_header' after actions in variations list

reuvenkarasik opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:
Added a new action called `woocommerce_variation_header` to the variations list in product editing screen, in the variation header (h3 tag) after all the actions (e.g. remove, sort, etc.).

This allows developers to add their own actions or information to the variation list.

I took inspiration from [the `woocommerce_variation_options` action](https://github.com/woocommerce/woocommerce/blob/8fd34983a5504a5dd428707dbb083d3a133f0f2c/includes/admin/meta-boxes/views/html-variation-admin.php#L96) in the same file.

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Added `woocommerce_variation_header` hook in variations list

reuvenkarasik committed:
Added action 'woocommerce_variation_header' action

kinging123 commented:
Closes #21362

claudiosanches commented on includes/admin/meta-boxes/views/html-variation-admin.php:
```diff
@@ -48,6 +48,17 @@
+		/**
+		 * woocommerce_variation_header action.
+		 *
+		 * @since 2.5.0
```
This version is incorrect, should be `3.6.0`.

claudiosanches commented on includes/admin/meta-boxes/views/html-variation-admin.php:
```diff
@@ -48,6 +48,17 @@
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param array   $variation_data
```
Missing description for `$variation_data`.

claudiosanches commented on includes/admin/meta-boxes/views/html-variation-admin.php:
```diff
@@ -48,6 +48,17 @@
+		 * @since 2.5.0
+		 *
+		 * @param array   $variation_data
+		 * @param WP_Post $variation
```
Missing description for `$variation`.

kinging123 replied:
Wasn't sure what to write here, I believe it's self-explanatory - `WP_Post $variation`.

Also, none of the other places in this file had a description for this param, even in the file header block.

Can you suggest a description, or should we leave it as is?

lkraav replied:
For self-explanatory variables I've written just ".", to indicate that there's nothing more to say and it's not a missed comment for code quality tools.

kinging123 replied:
Resolved in a9afc72

claudiosanches reviewed (changes requested):
Thanks for your contribution.
There is a few minor changes required before merge.

reuvenkarasik committed:
Fixed version and removed $variation_data array

Fixed version and removed $variation_data array (since it's deprecated) from filter `woocommerce_variation_header`

lkraav commented:
Thanks a ton. We're applying this patch on staging now and seeing how it works out. Really hoping it makes the 3.6.0  milestone.

Variation list has been neglected for a surprisingly long while. Playing the guessing game of "what's behind this door" must not have bothered enough store admins.. or is there a helper plugin I missed in the wild?

lkraav commented on includes/admin/meta-boxes/views/html-variation-admin.php:
```diff
@@ -48,6 +48,16 @@
 		?>
 		<input type="hidden" name="variable_post_id[<?php echo esc_attr( $loop ); ?>]" value="<?php echo esc_attr( $variation_id ); ?>" />
 		<input type="hidden" class="variation_menu_order" name="variation_menu_order[<?php echo esc_attr( $loop ); ?>]" value="<?php echo esc_attr( $variation_object->get_menu_order( 'edit' ) ); ?>" />
+		
```
There's a hidden whitespace error here, empty lines should not contain any characters.

kinging123 replied:
Resolved in a9afc72

lkraav commented on includes/admin/meta-boxes/views/html-variation-admin.php:
```diff
@@ -48,6 +48,16 @@
+		 *
+		 * @param WP_Post $variation
+		 */
+		do_action( 'woocommerce_variation_header', $variation ); ?>
```
Closing `?>` should probably go underneath `do_action()` to follow upstream coding standards.

kinging123 replied:
Resolved in a9afc72

reuvenkarasik committed:
Updated html-variation-admin.php to comply with CS

Based on @ikraav 's proposals

kinging123 commented:
I haven't seen any helper plugins that work with the variations list...
I guess most websites use variations for simple things like Color, Size, etc. So usually they'd have only a few variations and it wouldn't be so hard to maintain.

The problem is when you have dozens and hundreds of variations for each product, like the website I'm working on right now, and then maintaining it becomes a hard task.

lkraav reviewed (commented):
@lkraav not @ikraav tho
